src/main/java/org/jabref/logic/importer/fetcher/IEEE.java

```diff
@@ -205,8 +207,24 @@ public Parser getParser() {
                 JSONArray results = jsonObject.getJSONArray("articles");
                 for (int i = 0; i < results.length(); i++) {
                     JSONObject jsonEntry = results.getJSONObject(i);
-                    BibEntry entry = parseJsonRespone(jsonEntry, preferences.getKeywordSeparator());
-                    entries.add(entry);
+                    BibEntry entry = parseJsonResponse(jsonEntry, preferences.getKeywordSeparator());
+                    boolean addEntry;
+                    // In case entry has no year, add it
+                    // In case an entry has a year, check if its in the year range
+                    // The implementation uses some Java 8 Optional magic to implement that
+                    if (entry.hasField(StandardField.YEAR)) {
```

Comment:
entry.getField is already an optional, so no need to check if it's avaiable

Reply:
Pardon? I decided to put the non-negated form in the if condition. The else part is as follows:

```java
                    } else {
                        addEntry = true;
                    }
```

I can swap the branches:

```java
if (!entry.hasField(StandardField.YEAR)) {
  addEntry = true;
} else {
  // some magic happens here
}
```